### ISSUE TITLE
Override cache key per request

### ISSUE THREAD

**Issue by boedy** [RasCarlito/axios-cache-adapter#220]
This issue is similar or might be related to https://github.com/RasCarlito/axios-cache-adapter/issues/99. I'm not able to override the cache key per request:

```js
const options = {
    cache: {
        debug: true,
        key: (req) => {
            console.log('This method is not called');
        },
    },
};

axiosInstance.get('/resources', options).then((resp) => {
    console.log(resp.request.fromCache);
    return resp.data;
});
```

It keeps using the default key.

Here is an example where this bug is exposed (thank you @jahead for the example code from other issue ;) )
https://codesandbox.io/s/axios-cache-api-testing-forked-xr64f?file=/src/App.js

**Comment by jahead** [comment]
Let me have a look. What version of axios are you using?

This might be fixed with the update to axios 20. Just waiting on @RasCarlito for merging #215

**Comment by jahead** [comment]
https://codesandbox.io/s/axios-cache-api-testing-forked-y146g?file=/src/App.js
This solution works. 

```
const cache2 = setupCache({
  maxAge: 1 * 60 * 1000,
  debug: true,
  key: (req) => {
    console.log("Using custom cache key");
    return "custom-cache-key";
  }
});

const options = {
  adapter: cache2.adapter
};

axiosInstance.get('/resources', options).then((resp) => {
    console.log(resp.request.fromCache);
    return resp.data;
});
```

**Comment by boedy** [comment]
Thanks @jahead! I'm assuming your proposed solution is a workaround for now? Initialising a full adapter seems a bit much. I'll have a look at your PR and maybe just use that instead.

**Comment by RasCarlito** [comment]
Should now be fixed with latest release https://github.com/RasCarlito/axios-cache-adapter/releases/tag/2.6.0

**Comment by boedy** [comment]
@RasCarlito I just tried the 2.7.0 release. Unfortunately this issue still exists. Setting a `key` override has no affect.

**Comment by RasCarlito** [comment]
@boedy I'll have a look ASAP 🙂

**Comment by RasCarlito** [comment]
Just to let you know, I'm working on a fix for this. The cache `key` option was not included in the request override. I have an issue to fix on the tests and I will post the PR and probably make a release right after because I have another PR ready to be merged.

**Comment by RasCarlito** [comment]
Ok now it should really be fixed in latest release https://github.com/RasCarlito/axios-cache-adapter/releases/tag/2.7.1
